app/src/main/assets/debug/DebugActivity.java

```diff
@@ -60,17 +64,21 @@ protected void onCreate(Bundle savedInstanceState) {
             }
         }
 
-        AlertDialog dialog = new AlertDialog.Builder(this)
-                .setTitle("An error occurred")
-                .setMessage(madeErrorMessage)
-                .setPositiveButton("End Application", new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        finish();
-                    }
-                })
-                .create();
-        dialog.show();
-        ((TextView) dialog.findViewById(android.R.id.message)).setTextIsSelectable(true);
+        setTitle(getTitle() + " Crashed");
+        setContentView(contentView(this, madeErrorMessage));
+    }
+
+    View contentView(Context context, String errorMessage) {
```

Comment:
Why a separate method? Why pass over `Context` in a non-static method?

Reply:
??? The method just returns the view, what's that got to do with non static

I put it in a method without any specific reason. just keeping the view buildings out of onCreate

Reply:
It'd just be cleaner without an unnecessary extra method. The parameter is redundant too, since `this` can be used (it's not a static method).